eventrouter.go

```diff
@@ -144,6 +151,31 @@ func (er *EventRouter) addEvent(obj interface{}) {
 func (er *EventRouter) updateEvent(objOld interface{}, objNew interface{}) {
 	eOld := objOld.(*v1.Event)
 	eNew := objNew.(*v1.Event)
+
+	// Detect if the Informer is in resync
+	// In a re-sync previous events are re-submitted as updateEvents, which means  ResourceVersions will match between the eOld and eNew.
+	// Since there could be legimtate k8s reasons to re-emit an event, we also check to see if the last
+	// time vectors reset was at least X time ago as indicated by the resync interval.
+	if eOld.ResourceVersion == eNew.ResourceVersion {
+		reset := false
+
+		if er.firstReset || er.lastReset.IsZero() || time.Since(er.lastReset) >= viper.GetDuration("resync-interval") {
```

Comment:
If I am reading this correctly don't firstReset and lastReset.IsZero() essentially represent the same thing? LastReset being the zero epoch means it has never been reset.